app/src/main/java/org/android/go/sopt/MainActivity.kt

```diff
@@ -1,11 +1,87 @@
+import androidx.appcompat.app.AppCompatActivity
+import org.android.go.sopt.databinding.ActivityMainBinding
+import com.google.android.material.snackbar.Snackbar
 
 class MainActivity : AppCompatActivity() {
+    lateinit var binding: ActivityMainBinding //바인딩 객체 선언 (자동생성됨)
+    private lateinit var getResultText:ActivityResultLauncher<Intent>
+    lateinit var answer_nickname : String
+    lateinit var answer_tmi : String
+
+    // 화면 터치 시 키보드 내리기
+    override fun dispatchTouchEvent(ev: MotionEvent): Boolean {
+        val imm: InputMethodManager =
+            getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.hideSoftInputFromWindow(currentFocus?.windowToken, 0)
+        return super.dispatchTouchEvent(ev)
+    }
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater) // 액티비티에서 사용할 바인딩 클래스의 객체화
+        setContentView(binding.root) // getRoot 메서드로 레이아웃 최상단 뷰를 액티비티에 표시 합니다
+        var answer_id : String? = null
+        var answer_pw : String? = null
+
+
+        //Activity Result에 콜백 등록
+        getResultText = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == RESULT_OK) {
+                answer_id = result.data?.getStringExtra("data_id")?:"" // 얘는 엘비스 연산자 nullable하다는 뜻 -> 얘만 써
+                answer_pw = result.data?.getStringExtra("data_pw")?:""
+                answer_nickname = result.data?.getStringExtra("data_nickname")?:"" // !! -> nonNullable하도록 만들어 줌
+                answer_tmi = result.data?.getStringExtra("data_tmi")?:""
+            }
+            else{
+                Snackbar.make(binding.root, "회원 정보가 없습니다", Snackbar.LENGTH_SHORT).show()
+            }
+        } // 여기까지
+
+        // 로그인 버튼 클릭시
+        binding.btLogin.setOnClickListener {
+            val input_id = binding.editId.text.toString()
+            val input_pw = binding.editPw.text.toString()
+            // 입력된 정보가 다를 경우
+
+            when {
+                (answer_id == null) || (answer_pw == null)->{
+                Snackbar.make(binding.root, "회원가입을 진행해주세요.", Snackbar.LENGTH_SHORT).show()
+                }
+                (input_id != answer_id) || (input_pw != answer_pw)->{
+                    Snackbar.make(binding.root, "아이디와 비밀번호를 확인해주세요.", Snackbar.LENGTH_SHORT).show()
+                }
+                else->{
+                    Snackbar.make(binding.root, "로그인 합니다", Snackbar.LENGTH_SHORT).show()
+                    val intent = Intent(this, IntroduceActivity::class.java) // 자기소개로 이동
+                    intent.putExtra("data_id", answer_id)
+                    intent.putExtra("data_pw", answer_pw)
+                    intent.putExtra("data_nickname", answer_nickname)
+                    intent.putExtra("data_tmi", answer_tmi)
```

Comment:
좀 더 코틀린스럽게 바꿔보자면~
```suggestion
                    val intent = Intent(this, IntroduceActivity::class.java).apply {
                        putExtra("data_id", answer_id)
                        putExtra("data_pw", answer_pw)
                        putExtra("data_nickname", answer_nickname)
                        putExtra("data_tmi", answer_tmi)
                    }
```
의진님의 코드에서도 많이 봤을텐데 `apply`, `let` 등등은 **scope function** 이라고 합니다! 
코틀린만의 특징이기도 하고, 쓰면 가독성이 좋아지고 코드 길이도 줄일 수 있어서 scope function 활용을 습관화하는게 좋습니다~!
[공식문서](https://kotlinlang.org/docs/scope-functions.html)도 놓고 갑니당